ember-file-upload/addon/queue.ts

```diff
@@ -204,9 +204,10 @@ export default class Queue {
       }
 
       const selectedFiles: UploadFile[] = [];
+      let index = 0;
 
       for (const file of files) {
-        if (named.filter && !named.filter?.(file)) {
+        if (named.filter && !named.filter?.(file, files, index++)) {
```

Comment:
`files` will be a FileList rather than `File[]`. Suggest something like this:

```js
      const { files: fileList } = event.target as HTMLInputElement;
      if (!fileList) {
        return;
      }

      const files = Array.from(fileList);
      const selectedFiles: UploadFile[] = [];

      for (const file of files) {
        if (named.filter && !named.filter?.(file, files, files.indexOf(file))) {
          continue;
        }
```